File: DesignPatternsExercises/java/FlyweightDesignPattern/FlyweightDesignPattern1/Exercise/FloweringCherry.java
public class FloweringCherry __________(1)__________ Tree {
    private String flowerColor;

    public FloweringCherry() {
        __________(2)__________ (true, false, 20);
        flowerColor = "Pink";
    }

    @Override
    public void changeProperty() {
        System.out.println("Changing Flowering Cherry property Flower Color to Violet\n");
        setFlowerColor("Violet");
    }

    public String getFlowerColor() {
        return flowerColor;
    }

    public void setFlowerColor(String flowerColor) {
        this.flowerColor = flowerColor;
    }

    @Override
    public String toString() {
        return "Flowering Cherry ~~ Height: " + getHeight() + "  Flowers: " + hasFlowers() +
                "  Fruits: " + hasFruits() + "  Flower Color: " + getFlowerColor();
    }
}

File: DesignPatternsExercises/ruby/flyweight_design_pattern/flyweight_design_pattern1/exercise/flowering_cherry.rb
require_relative 'tree'

class FloweringCherry < __________(1)__________
  def __________(2)__________
    __________(3)__________(true, false, 20)
    @flower_color = :Pink
  end

  def __________(4)__________
    puts 'Changing Flowering Cherry property Flower Color to Violet'
    puts
    @flower_color = :Violet
  end

  def to_string
    "Flowering Cherry ~~ Height: #{@height}  Flowers: #{@has_flowers}  Fruits: #{@has_fruits}  Flower Color: #{@flower_color}"
  end
end
